main_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"syscall"
+	"testing"
+	"time"
+
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+)
+
+var kernelImagePath = "./vmlinux"
+var rootDrivePath = "/dev/null"
+
+func init() {
+	if v := os.Getenv("KERNELIMAGE"); len(v) != 0 {
+		kernelImagePath = v
+	}
+
+	if v := os.Getenv("ROOTDRIVE"); len(v) != 0 {
+		rootDrivePath = v
+	}
+}
+
+func TestFireCTL(t *testing.T) {
+	const firectlName = "./firectl"
+	if _, err := os.Stat(firectlName); os.IsNotExist(err) {
+		t.Skip("Missing firectl binary")
+	}
+
+	if _, err := os.Stat(kernelImagePath); os.IsNotExist(err) {
+		t.Skip("Missing kernel image")
+	}
+
+	if _, err := os.Stat(rootDrivePath); os.IsNotExist(err) {
+		t.Skip("Missing root drive")
+	}
+
+	ctx := context.Background()
+	ctlCtx, cancelFn := context.WithTimeout(ctx, 500*time.Millisecond)
+	defer cancelFn()
+
+	const socketpath = "./integration_test.sock"
+	firectlArgs := []string{
+		"-s",
+		socketpath,
+		"--kernel",
+		kernelImagePath,
+		"--root-drive",
+		rootDrivePath,
+	}
+	cmd := exec.CommandContext(ctlCtx, firectlName, firectlArgs...)
+	if err := cmd.Start(); err != nil {
+		t.Fatalf("Failed to run command: %v", err)
+	}
+
+	defer func() {
+		if err := os.Remove(socketpath); err != nil {
+			t.Log("Failed to remove socket path file")
+		}
+
+		if err := cmd.Process.Signal(syscall.SIGQUIT); err != nil {
+			t.Log("Failed to kill firectl process")
+		}
+	}()
+
+	tr := &http.Transport{
+		Dial: func(proto, addr string) (net.Conn, error) {
+			return net.Dial("unix", socketpath)
+		},
+	}
+	client := &http.Client{Transport: tr}
+
+	valid := false
+	interval := 50 * time.Millisecond
+	payload := &models.InstanceInfo{}
+	for i := 0; i < 10; i++ {
+		time.Sleep(interval)
```

Comment:
Do we always want to sleep for 50ms even on the first iteration?

Reply:
Reasoning for this was that the continue block would also need to sleep meaning there would be two locations, one in the if block and another at the end of the for loop. I will go ahead a change it.